File: day_01/main.py
from collections import defaultdict

col1, col2 = [], []

col2_hash = defaultdict(int)


with open("input.txt") as f:
    for line in f:
        v1, v2 = [int(v.strip()) for v in line.split("   ")]
        col1.append(v1)
        col2.append(v2)
    for v in col2:
        col2_hash[v] += 1
    total = sum(v * col2_hash[v] for v in col1)
    print(total)
    #     col1 = sorted(col1)
    #     col2 = sorted(col2)
    # total = sum(abs(v1 - v2) for v1, v2 in zip(col1, col2))
    # print(total)

File: day_01/main.cpp
#include <algorithm>
#include <cstdlib>
#include <fstream>
#include <iostream>
#include <ranges>
#include <sstream>
#include <string>
#include <vector>

int main() {
  std::ifstream in{"input.txt"};
  if (!in) {
    return 1;
  }

  std::vector<int> c1, c2;
  int n1, n2;
  while (in >> n1 >> n2) {
    c1.push_back(n1);
    c2.push_back(n2);
  }

  std::sort(c1.begin(), c1.end());
  std::sort(c2.begin(), c2.end());

  int total{0};
  for (auto [n1, n2] : std::views::zip(c1, c2)) {
    total += abs(n2 - n1);
  }

  std::cout << total << std::endl;

  return 0;
}
